packages/flutter/lib/src/rendering/editable_line.dart

```diff
@@ -192,10 +203,19 @@ class RenderEditableLine extends RenderBox {
     return _textPainter.getPositionForOffset(globalToLocal(globalPosition).toOffset());
   }
 
+  /// Returns the Rect in local coordinates for the caret at the given text
+  /// position.
+  Rect getLocalRectForCaret(TextPosition caretPosition) {
+    double lineHeight = constraints.constrainHeight(_preferredLineHeight);
+    Offset caretOffset = _textPainter.getOffsetForCaret(caretPosition, _caretPrototype);
+    // This rect is the same as _caretPrototype but without the vertical padding.
+    return new Rect.fromLTWH(0.0, 0.0, _kCaretWidth, lineHeight).shift(caretOffset + _paintOffset);
+  }
```

Comment:
changed the caret rect I use here, dropping the vertical padding so that we scroll all the way to the edge when it moves.